packages/@aws-cdk/region-info/lib/region-info.ts

```diff
@@ -0,0 +1,73 @@
+import os = require('os');
+import path = require('path');
+import process = require('process');
+import { IRegionInfo } from './iregion-info';
+import builtIns = require('./static.generated');
+import { loadUserData } from './user-data';
+
+const CDK_REGION_INFO_PATH_VARIABLE = 'CDK_REGION_INFO_PATH';
+const CDK_REGION_INFO_PATH_DEFAULT = path.resolve(os.homedir(), '.cdk', 'region-info.json');
+
+/**
+ * Allows accessing region information configured in this process. The library provides built)in configuration for
+ * AWS region information, however newer regions might be missing unless they have been properly configured in the
+ * system. In such cases, a user can provide their own configuration entries in a JSON document thait is located in
+ * the path denoted by `RegionInfo.userDataPath`.
+ */
+export class RegionInfo {
+  /**
+   * The path from which user-defined region information is loaded. User-defined information always has precedence
+   * over built-in data (this guarantees consistent behavior across updates to this library). The default value for
+   * this property is `~/.cdk/region-info.json`, or the value of the environment variable `CDK_REGION_INFO_PATH` if it
+   * is set.
+   */
+  public static get userDataPath(): string {
```

Comment:
as mentioned, I don't think this library should be in the business of loading files from disk

Reply:
There is some disagreement here, and I believe @rix0rrr  is right that we should have a way to load from a "standard" file, otherwise bootstrapping with missing regions is a pain.

Reply:
Can you explain why is it a pain to do: `fs.readFileSync('config.json')`?

Reply:
Just trying to keep things as simple as possible and avoid over engineering